src/systems/shadow-system.js

```diff
@@ -90,46 +88,44 @@ function resizeShadowCameraFrustum(light, boundingBox) {
 
 export class ShadowSystem {
   constructor(sceneEl) {
-    this.needsUpdate = false;
-    this.dynamicShadowsEnabled = window.APP.store.state.preferences.enableDynamicShadows;
     this.sceneEl = sceneEl;
-    this.onEnvironmentSceneLoaded = this.onEnvironmentSceneLoaded.bind(this);
-    this.sceneEl.addEventListener("environment-scene-loaded", this.onEnvironmentSceneLoaded);
     this.shadowCameraBoundingBox = new THREE.Box3();
+    this.previousShadowsEnabled = this.realtimeShadowsEnabled = null;
+    window.APP.store.addEventListener("statechanged", this.updatePrefs.bind(this));
+    this.updatePrefs();
   }
 
-  onEnvironmentSceneLoaded({ detail: environmentObject3D }) {
-    this.environmentObject3D = environmentObject3D;
-    this.needsUpdate = true;
-    this.sceneEl.renderer.shadowMap.autoUpdate = this.dynamicShadowsEnabled;
+  updatePrefs() {
+    this.realtimeShadowsEnabled =
```

Comment:
It might be a bit confusing if you are in low material quality but check this box. Since we will not have it off by default maybe we should just directly follow the preference even in low quality? Another option would be to hide/disable the preference when in low quality mode.

Reply:
I feel the same. And I already opened an issue for it #5210
I don't really want to change the behavior of low quality mode in this PR. Let's make another PR if needed.